src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -106,17 +112,35 @@ LinalgBackendBase* infer_backend(const Container<T>& a, const Container<T>& b)
  * and a shallow-copy of SGVector with vector on CPU if GPU backend not available
  */
 template <typename T>
-SGVector<T> to_gpu(const SGVector<T>& vector)
+void to_gpu(SGVector<T>& a, SGVector<T>& b)
 {
```

Comment:
@vigsterkr I just checked a bit, and most of shogun is not thread safe at all. So I wonder whether we really should care about this too much at this point in time. I mean we should definitely have a go on making operations thread safe (along with proper const-ing), but for now I feel that, yes, it is user responsibility. Even for GPU transfers.
I mean we can still keep this patch here, but I think there should be some more systematic effort to make shogun base lib thread safe.